code/modules/clothing/chameleon.dm

```diff
@@ -760,3 +763,167 @@
+	chameleon_projectile_vars = list()
+
+	if(chambered)
+		chambered.firing_effect_type = initial(chambered.firing_effect_type)
+
+	fire_sound = initial(fire_sound)
+	burst_size = initial(burst_size)
+	fire_delay = initial(fire_delay)
+	inhand_x_dimension = initial(inhand_x_dimension)
+	inhand_y_dimension = initial(inhand_y_dimension)
+
+	QDEL_NULL(chambered.loaded_projectile)
+	chambered.newshot()
+
+/**
+ * Description: Sets what gun we should be mimicking.
+ * Arguements: [obj/item/gun/gun_to_set (the gun we're trying to mimic), passthrough (whether or not we're setting ammo too)]
+ */
+/obj/item/gun/energy/laser/chameleon/proc/set_chameleon_gun(obj/item/gun/gun_to_set, passthrough = TRUE)
+	if(!istype(gun_to_set))
+		stack_trace("[gun_to_set] is not a valid typepath.")
+		return FALSE
+
+	fire_sound = gun_to_set.fire_sound
+	burst_size = gun_to_set.burst_size
+	fire_delay = gun_to_set.fire_delay
+	inhand_x_dimension = gun_to_set.inhand_x_dimension
+	inhand_y_dimension = gun_to_set.inhand_y_dimension
+
+	if(passthrough)
+		if(istype(gun_to_set, /obj/item/gun/ballistic))
+			var/obj/item/gun/ballistic/ball_gun = gun_to_set
+			var/obj/item/ammo_box/ball_ammo = new ball_gun.mag_type(gun_to_set)
+			qdel(ball_gun)
+
+			if(!istype(ball_ammo) || !ball_ammo.ammo_type)
+				qdel(ball_ammo)
+				return FALSE
+
+			var/obj/item/ammo_casing/ball_cartridge = new ball_ammo.ammo_type(gun_to_set)
+			set_chameleon_ammo(ball_cartridge)
+
+		else if(istype(gun_to_set, /obj/item/gun/magic))
+			var/obj/item/gun/magic/magic_gun = gun_to_set
+			var/obj/item/ammo_casing/magic_cartridge = new magic_gun.ammo_type(gun_to_set)
+			set_chameleon_ammo(magic_cartridge)
+
+		else if(istype(gun_to_set, /obj/item/gun/energy))
+			var/obj/item/gun/energy/energy_gun = gun_to_set
+			if(islist(energy_gun.ammo_type) && energy_gun.ammo_type.len)
+				var/obj/item/ammo_casing/energy_cartridge = energy_gun.ammo_type[1]
+				set_chameleon_ammo(energy_cartridge)
+
+		else if(istype(gun_to_set, /obj/item/gun/syringe))
+			var/obj/item/ammo_casing/syringe_cartridge = new /obj/item/ammo_casing/syringegun(src)
+			set_chameleon_ammo(syringe_cartridge)
+
+/**
+ * Description: Sets the ammo type our gun should have.
+ * Arguements: [obj/item/ammo_casing/cartridge (the ammo_casing we're trying to copy), passthrough (whether or not we're the loaded projectile too)]
+ */
+/obj/item/gun/energy/laser/chameleon/proc/set_chameleon_ammo(obj/item/ammo_casing/cartridge, passthrough = TRUE)
+	if(!istype(cartridge))
+		stack_trace("[cartridge] is not a valid typepath.")
+		return FALSE
+
+	if(passthrough)
+		var/obj/projectile/proj = cartridge.loaded_projectile
+		set_chameleon_projectile(proj)
+
+/**
+ * Description: Sets the current projectile variables for our chameleon gun.
+ * Arguements: [obj/projectile/cham_projectile (the projectile we're trying to copy)]
+ */
+/obj/item/gun/energy/laser/chameleon/proc/set_chameleon_projectile(obj/projectile/cham_projectile)
```

Comment:
Full word: `chameleon_projectile`

Reply:
You should probably rename it to `template_projectile` or `projectile_to_copy`, though. It was not obvious to me.